### PR TITLE
change play&seek position to milliseconds and add testing example

### PR DESCRIPTION
**Pull request by catoldcui**
For resolving [#83](https://github.com/luanpotter/audioplayers/issues/83).

### CODE REVIEW

**Comment by OperKH** [comment]
LGTM

**Comment by erickzanardo** [comment]
Thanks for the PR, will be testing this today after work!

**Comment by luanpotter** [comment]
Thanks a lot, @catoldcui !

I'd make one change though, since we are going to support now millis to the precision of millis, I'd change the channel API to actually receive and integer instead of double, so to avoid unnecessary casts and rounding and also not to give the impression that we would accept half a millisecond, for example.

File: android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java [hunk 1]
```diff
@@ -175,7 +175,7 @@ void pause() {
     @Override
     void seek(double position) {
```
Comment (luanpotter) [inline android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java]: I'd change this api to int

File: android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java [hunk 2]
```diff
@@ -175,7 +175,7 @@ void pause() {
...
         if (this.prepared)
-            this.player.seekTo((int) (position * 1000));
+            this.player.seekTo((int) position);
```
Comment (luanpotter) [inline android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java]: then no cast here

File: android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java [hunk 3]
```diff
@@ -175,7 +175,7 @@ void pause() {
...
         else
             this.shouldSeekTo = position;
```
Comment (luanpotter) [inline android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java]: shouldSeekTo should be an int as well

File: android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java [hunk 4]
```diff
@@ -192,7 +192,7 @@ public void onPrepared(final MediaPlayer mediaPlayer) {
             ref.handleIsPlaying(this);
         }
         if (this.shouldSeekTo >= 0) {
-            this.player.seekTo((int) (this.shouldSeekTo * 1000));
+            this.player.seekTo((int) this.shouldSeekTo);
```
Comment (luanpotter) [inline android/src/main/java/xyz/luan/audioplayers/WrappedMediaPlayer.java]: no cast here!

File: ios/Classes/AudioplayersPlugin.m [hunk 5]
```diff
@@ -66,12 +66,12 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
                         result(0);
                     int isLocal = [call.arguments[@"isLocal"]intValue] ;
                     float volume = (float)[call.arguments[@"volume"] doubleValue] ;
-                    double seconds = call.arguments[@"position"] == [NSNull null] ? 0.0 : [call.arguments[@"position"] doubleValue] ;
+                    double milliseconds = call.arguments[@"position"] == [NSNull null] ? 0.0 : [call.arguments[@"position"] doubleValue] ;
```
Comment (luanpotter) [inline ios/Classes/AudioplayersPlugin.m]: int

File: ios/Classes/AudioplayersPlugin.m [hunk 6]
```diff
@@ -100,9 +100,9 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
                     if (!call.arguments[@"position"]) {
                       result(0);
                     } else {
-                      double seconds = [call.arguments[@"position"] doubleValue];
-                      NSLog(@"Seeking to: %f seconds", seconds);
-                      [self seek:playerId time:CMTimeMakeWithSeconds(seconds,1)];
+                      double milliseconds = [call.arguments[@"position"] doubleValue];
```
Comment (luanpotter) [inline ios/Classes/AudioplayersPlugin.m]: int

File: lib/audioplayers.dart [hunk 7]
```diff
@@ -159,12 +159,12 @@ class AudioPlayer {
     Duration position: Duration.zero,
     bool respectSilence: false,
   }) async {
-    final double positionInSeconds = position == null ? null : position.inSeconds.toDouble();
+    final double positionInMilliseconds = position == null ? null : position.inMilliseconds.toDouble();
```
Comment (luanpotter) [inline lib/audioplayers.dart]: no need for toDouble here either

File: lib/audioplayers.dart [hunk 8]
```diff
@@ -215,8 +215,7 @@ class AudioPlayer {
 
   /// Move the cursor to the desired position.
   Future<int> seek(Duration position) {
-    double positionInSeconds = position.inMicroseconds / Duration.microsecondsPerSecond;
-    return _invokeMethod('seek', {'position': positionInSeconds});
+    return _invokeMethod('seek', {'position': position.inMilliseconds.toDouble()});
```
Comment (luanpotter) [inline lib/audioplayers.dart]: nor here

**Comment by catoldcui** [comment]
Make sense. 

So please let know if you finish this task. I'll rebase and change all the cast you mentioned. And PR again.

**Comment by erickzanardo** [comment]
I will merge this, and change the casts on master

**Comment by adriancmurray** [comment]
I'm using this for seeking and on iOS on 0.11.0 (I can't use the most recent version due to a conflict with path_provider) it's not seeking to the correct millisecond. It's going to the lowest second. Isn't version 0.11.0 supposed to work with milliseconds?

**Pull request merged**